### ISSUE TITLE
Add support for Oblivious DoH

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/dnsproxy#332]
See the original feature request here:
https://github.com/AdguardTeam/AdGuardHome/issues/2406

The task consists of the following parts:

* [ ] Support ODoH on the client-side. This is the most important one as for both DnsLibs / AdGuard Home it only makes sense to have ODoH on the client-side.
* [ ] Add support for ODoH to [dnsstamps](https://github.com/ameshkov/dnsstamps)
* [ ] Add support for ODoH server-side. Nice-to-have, but we will probably only need it in dnsproxy.

Once the work is finished, we'll need to open a corresponding issue in DnsLibs.

**Comment by Sn0whax** [comment]
This is a shame, dev needs help and no one wants/knows how to implement this. :(

**Comment by rahilarious** [comment]
looking forward to this
